### PR TITLE
Endpoint for EIP-4881 Deposit Contract Snapshot

### PR DESCRIPTION
**Pull request by ethDreamer**
Adds the endpoint for downloading the [`EIP-4881`](https://eips.ethereum.org/EIPS/eip-4881#specification) Deposit Tree Snapshot during weak subjectivity sync.

### CODE REVIEW

**Review by michaelsproul — DISMISSED** [review]
LGTM once merge conflicts are addressed

File: apis/beacon/deposit_snapshot.yaml
Comment (arnetheduck) [inline apis/beacon/deposit_snapshot.yaml:31]: we use 404 for when the "thing" is known not to exist (for example an empty slot) - a different code would probably be useful here for ephemeral error conditions (503/syncing for example)

File: apis/beacon/deposit_snapshot.yaml
Comment (arnetheduck) [inline apis/beacon/deposit_snapshot.yaml:1]: should there be support for downloading historical snapshots? ie when checkpoint-syncing, one will want to have a matching snapshot to the state that's being synced, which may or may not (due to race conditions or checkpoint age choice) be the same as the snapshot being served by "default"

Comment (rolfyone) [reply apis/beacon/deposit_snapshot.yaml:1]: I think the main idea is just to get a starting point rather than download the whole deposit log, so I think there's probably not a need currently to get historical snapshots...

**Comment by rolfyone** [comment]
resolved conflicts. Let me know if you're interested in continuing discussion or happy to merge @arnetheduck

File: apis/beacon/deposit_snapshot.yaml
Comment (rolfyone) [reply apis/beacon/deposit_snapshot.yaml:31]: I do tend to agree 404 isn't ideal (a server with a route that is missing is also 404) - even if it was a 400, or potentially 503 might work... I'll see if i can get some more opinions on this one.

**Comment by rolfyone** [comment]
There's been no further feedback in more than 10 days. resolved conflicts again and will merge.

**Review by rolfyone — APPROVED** [review]
LGTM

**Pull request merged**
